h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -45,6 +47,7 @@
 
   String[] _synthetic_column_names; // Columns with constant values to be added to parsed Frame
   String[][] _synthetic_column_values; // For each imported file contains array of values for each synthetic column
+  byte _synthetic_column_type = Vec.T_STR; // By default, all synthetic columns are treated as strings
```

Comment:
Just a question: Isn't it better to treat synthetic columns as categoricals? The cardinality of these columns shouldn't be high.

Reply:
This is just a safe default - the whole parsing module is a big spaghetti. We treat it as categorical - if pratition_by parameter is passed, those synthetic columns will always be categorical.

Reply:
ok, thanks for explanation!